src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -1139,19 +1116,45 @@ namespace AppInstaller::Repository
                         }
                         catch (...)
                         {
+                            LOG_CAUGHT_EXCEPTION();
                             AICLI_LOG(Repo, Warning, << "Failed to update source: " << source.get().Name);
                             result.SourcesWithUpdateFailure.emplace_back(source);
                         }
                     }
 
-                    aggregatedSource->AddAvailableSource(CreateSourceFromDetails(source, progress));
+                    std::shared_ptr<ISource> openedSource;
+
+                    try
+                    {
+                        openedSource = CreateSourceFromDetails(source, progress);
+                    }
+                    catch (...)
+                    {
+                        LOG_CAUGHT_EXCEPTION();
+                        AICLI_LOG(Repo, Warning, << "Failed to open source: " << source.get().Name);
+                        openExceptionProxies.emplace_back(std::make_shared<OpenExceptionProxy>(source, std::current_exception()));

```

Comment:
>openExceptionProxies.emplace_back(std::make_shared<OpenExceptionProxy>(source, std::current_exception())); [](http://example.com/codeflow?start=24&length=106)

nit: I think we can just directly add the proxy to the aggregatedSource directly here instead of saving to a list and add all later?

Reply:
never mind, there's a none available check below